lib/commands/promote.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const chalk = require('chalk');
+const isMissingData = require('./validations').isMissingData;
+const max_promotions = 3;
+
+let elapsedInterval;
+let spamSeed = generateSeed()
+
+let promoteCount = 0;
+function interrupt() {
+    return promoteCount++ >= max_promotions;
+}
+// WARNING: Not cryptographically secure. Do not use any seeds generated by this generator to actually store any value.
+function generateSeed() {
+    const validChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ9'
+    return Array.from(new Array(81), (x, i) => validChars[Math.floor(Math.random() * validChars.length)]).join('')
+}
+
+function generateTransfer() {
+    return {
+        address: spamSeed,
+        value: 0,
+        message: '',
+        tag: ''
+    }
+}
+const setupPromoteCommand = (data, iotajs, refreshAccountData, vorpal) => {
+  vorpal
+    .command('promote <hash>', 'Promotes a pending transaction.  Provide an id for additional details.')
+    .option('-n <number>', 'Max number of transactions.  Default 10.')
+    .autocomplete({
+      data: () => {
+        vorpal.ui.input(`promote ${vorpal.ui.input().split(' ')[1].toUpperCase()}`);
+        if (!data.accountData) {
+          return [];
+        }
+        return data.accountData.transfers.map(b => b[0]).map(t => t.hash.slice(0, 6));
+      }
+    })
+    .action((args, callback) => {
+      if (isMissingData(['node'])) {
+        return callback();
+      }
+
+      var hash = args.hash;
+      if (hash.length < 81) {
+        if (!data.accountData) {
+          vorpal.log(chalk.red('Please provide a full transaction hash'));
+          return callback();
+        }
+        const transactionsMatchingHash = data.accountData.transfers
+          .map(b => b[0])
+          .filter(t => t.hash.indexOf(hash) !== -1);
+        if (transactionsMatchingHash.length === 0) {
+          vorpal.log(chalk.red('That hash does not match a transaction'));
+          return callback();
+        }
+        hash = transactionsMatchingHash[0].hash;
+      }
+
+      iotajs.api.getLatestInclusion([hash], (error, states) => {
+        var confirmFound = false;
+        if (error)
+        {
+            vorpal.log(chalk.red('Error checking confirmation status.'));
+            return callback();
+        }
+        states.forEach((confirmed, i) => {
+        	if (confirmed) {
+            confirmFound = true;
+          }
+        });
+        if (confirmFound)
+        {
+          vorpal.log(chalk.red('That transaction is already confirmed.'));
+          return callback();
+        } else {
+          vorpal.log(`Promoting transaction ${hash}.  This may take a few minutes.`);
+          const start = Date.now();
+          elapsedInterval = setInterval(() => {
+            process.stdout.write(`You've been waiting ${Math.floor((Date.now() - start)/1000)}s\r`);
+          });
+          iotajs.api.isPromotable(hash).then(state => {
+            if (state) {
+              promoteCount = 0;
```

Comment:
Need a `let` declaration here.  No `var`s down below please.

Reply:
I'm a bit confused about this request. We are promoting three times, in the `interrupt` function passed into `iota.api.promoteTransaction` it increments and returns false until we hit our max. What would you suggest the best way to handle this?

Reply:
I was asking that you add `let` on line 78.  It's an implied `var` now.  Do you still have questions about the promoteCount variable?  It's just a retry counter.  Is there something particular about it that is not making sense?

Reply:
Sorry, I've since understood. But I see now I have messed this branch up a bit by merging my master into it after the fact so I'll clean things up and redo a PR.

Reply:
Alright I'm still lost heh, apologies. I've declared the `let promoteCount` at line 9 so the `interrupt` function has access to it (otherwise I get a `ReferenceError: promoteCount is not defined`). I then set it to 0 each run of this command during a session otherwise it stays at 3 any time I try to promote a second time? How is it an implied var if I've globally declared the let or did you just not see that initial declaration? Should I not be using a global let?

Reply:
Ah, I didn't see it declared on line 9.  Thanks for pointing that out.